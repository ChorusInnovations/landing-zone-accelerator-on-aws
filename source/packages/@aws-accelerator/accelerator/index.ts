/**
 *  Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

export * from './lib/accelerator';
export * from './lib/config-repository';
export * from './lib/pipeline';
export * from './lib/stacks/accounts-stack';
export * from './lib/stacks/default-stack';
export * from './lib/stacks/dependencies-stack';
export * from './lib/stacks/network-1-stack';
export * from './lib/stacks/network-2-stack';
export * from './lib/stacks/network-3-stack';
export * from './lib/stacks/operations-stack';
export * from './lib/stacks/organizations-stack';
export * from './lib/stacks/pipeline-stack';
export * from './lib/stacks/security-audit-stack';
export * from './lib/stacks/security-stack';
export * from './lib/stacks/validate-stack';
export * from './lib/toolkit';
