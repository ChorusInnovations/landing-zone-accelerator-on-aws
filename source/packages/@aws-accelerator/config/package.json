{
  "name": "@aws-accelerator/config",
  "version": "0.0.0",
  "license": "Apache-2.0",
  "author": {
    "name": "Amazon Web Services",
    "url": "https://aws.amazon.com"
  },
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "scripts": {
    "cleanup": "tsc --build ./ --clean && rm -rf node_modules && rm -rf yarn.lock && rm -rf dist && rm -rf cdk.out",
    "cleanup:tsc": "tsc --build ./ --clean",
    "build": "tsc",
    "watch": "tsc -w",
    "test": "jest --coverage",
    "testreport": "jest --ci --coverage",
    "lint": "eslint --fix  --max-warnings 0 -c ../../../.eslintrc.json 'lib/**/*.{ts,tsx}' 'test/**/*.{ts,tsx}' --ignore-pattern \"*.d.ts\" ",
    "precommit": "eslint --max-warnings 0 -c ../../../.eslintrc.json 'lib/**/*.{ts,tsx}' 'test/**/*.{ts,tsx}' --ignore-pattern \"*.d.ts\" "
  },
  "devDependencies": {
    "@types/jest": "26.0.10",
    "@types/node": "16.9.1",
    "@typescript-eslint/eslint-plugin": "4.31.1",
    "@typescript-eslint/parser": "4.31.1",
    "eslint": "7.32.0",
    "eslint-config-prettier": "8.3.0",
    "eslint-config-standard": "16.0.3",
    "eslint-import-resolver-node": "0.3.2",
    "eslint-import-resolver-typescript": "2.0.0",
    "eslint-plugin-import": "2.19.1",
    "eslint-plugin-license-header": "0.2.0",
    "eslint-plugin-node": "11.1.0",
    "eslint-plugin-prettier": "4.0.0",
    "jest": "26.4.2",
    "prettier": "2.4.1",
    "ts-jest": "26.2.0",
    "typescript": "4.4.3"
  },
  "dependencies": {
    "fp-ts": "2.10.5",
    "io-ts": "2.2.16",
    "ip-num": "1.3.1"
  }
}
