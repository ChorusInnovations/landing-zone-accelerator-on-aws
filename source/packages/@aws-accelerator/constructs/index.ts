export * from './lib/aws-organizations/organizational-unit';
export * from './lib/aws-organizations/root-organizational-unit';
