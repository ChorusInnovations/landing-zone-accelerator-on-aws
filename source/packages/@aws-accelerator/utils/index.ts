export * from './lib/policy-replacements';
export * from './lib/throttle';
