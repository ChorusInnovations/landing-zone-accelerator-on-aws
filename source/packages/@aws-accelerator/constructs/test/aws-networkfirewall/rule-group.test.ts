/**
 *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

import * as cdk from 'aws-cdk-lib';

import { NfwRuleGroupRuleConfig } from '@aws-accelerator/config';

import { NetworkFirewallRuleGroup } from '../../lib/aws-networkfirewall/rule-group';

const testNamePrefix = 'Construct(NetworkFirewallPolicy): ';

//Initialize stack for resource configuration test
const stack = new cdk.Stack();

const ruleGroup: NfwRuleGroupRuleConfig = {
  rulesSource: {
    statefulRules: [
      {
        action: 'PASS',
        header: {
          destination: '10.0.0.0/16',
          destinationPort: 'ANY',
          direction: 'FORWARD',
          protocol: 'IP',
          source: '10.1.0.0/16',
          sourcePort: 'ANY',
        },
        ruleOptions: [
          {
            keyword: 'sid',
            settings: ['100'],
          },
        ],
      },
    ],
    rulesSourceList: undefined,
    statelessRulesAndCustomActions: undefined,
    rulesString: undefined,
  },
  ruleVariables: undefined,
  statefulRuleOptions: 'STRICT_ORDER',
};

new NetworkFirewallRuleGroup(stack, 'TestGroup', {
  name: 'TestGroup',
  capacity: 100,
  type: 'STATEFUL',
  ruleGroup: ruleGroup,
  tags: [],
});

/**
 * Network Firewall construct test
 */
describe('Network Firewall Rule Group', () => {
  /**
   * Number of Network Firewall Policy test
   */
  test(`${testNamePrefix} Network firewall rule group count test`, () => {
    cdk.assertions.Template.fromStack(stack).resourceCountIs('AWS::NetworkFirewall::RuleGroup', 1);
  });

  /**
   * Network firewall resource configuration test
   */
  test(`${testNamePrefix} Network firewall policy resource configuration test`, () => {
    cdk.assertions.Template.fromStack(stack).templateMatches({
      Resources: {
        TestGroupAF88660E: {
          Type: 'AWS::NetworkFirewall::RuleGroup',
          Properties: {
            Capacity: 100,
            RuleGroup: {
              RulesSource: {
                StatefulRules: [
                  {
                    Action: 'PASS',
                    Header: {
                      Destination: '10.0.0.0/16',
                      DestinationPort: 'ANY',
                      Direction: 'FORWARD',
                      Protocol: 'IP',
                      Source: '10.1.0.0/16',
                      SourcePort: 'ANY',
                    },
                    RuleOptions: [
                      {
                        Keyword: 'sid',
                        Settings: ['100'],
                      },
                    ],
                  },
                ],
              },
              StatefulRuleOptions: {
                RuleOrder: 'STRICT_ORDER',
              },
            },
            RuleGroupName: 'TestGroup',
            Tags: [
              {
                Key: 'Name',
                Value: 'TestGroup',
              },
            ],
            Type: 'STATEFUL',
          },
        },
      },
    });
  });
});
