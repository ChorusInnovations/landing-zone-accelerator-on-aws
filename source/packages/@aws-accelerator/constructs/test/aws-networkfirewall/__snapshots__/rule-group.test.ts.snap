// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Network Firewall Rule Group Construct(NetworkFirewallPolicy):  Snapshot Test 1`] = `
Object {
  "Resources": Object {
    "TestGroupAF88660E": Object {
      "Properties": Object {
        "Capacity": 100,
        "RuleGroup": Object {
          "RulesSource": Object {
            "StatefulRules": Array [
              Object {
                "Action": "PASS",
                "Header": Object {
                  "Destination": "10.0.0.0/16",
                  "DestinationPort": "ANY",
                  "Direction": "FORWARD",
                  "Protocol": "IP",
                  "Source": "10.1.0.0/16",
                  "SourcePort": "ANY",
                },
                "RuleOptions": Array [
                  Object {
                    "Keyword": "sid",
                    "Settings": Array [
                      "100",
                    ],
                  },
                ],
              },
            ],
          },
          "StatefulRuleOptions": Object {
            "RuleOrder": "STRICT_ORDER",
          },
        },
        "RuleGroupName": "TestGroup",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestGroup",
          },
        ],
        "Type": "STATEFUL",
      },
      "Type": "AWS::NetworkFirewall::RuleGroup",
    },
    "TestGroupStatelessB53F2424": Object {
      "Properties": Object {
        "Capacity": 100,
        "RuleGroup": Object {
          "RuleVariables": Object {
            "IPSets": Object {
              "HOME_NET": Object {
                "Definition": Array [
                  "10.0.0.0/16",
                ],
              },
            },
            "PortSets": Object {
              "HOME_NET": Object {
                "Definition": Array [
                  "80",
                  "443",
                ],
              },
            },
          },
          "RulesSource": Object {
            "StatelessRulesAndCustomActions": Object {
              "CustomActions": Array [
                Object {
                  "ActionDefinition": Object {
                    "PublishMetricAction": Object {
                      "Dimensions": Array [
                        Object {
                          "Value": "CustomValue",
                        },
                      ],
                    },
                  },
                  "ActionName": "CustomAction",
                },
              ],
              "StatelessRules": Array [
                Object {
                  "Priority": 100,
                  "RuleDefinition": Object {
                    "Actions": Array [
                      "aws:pass",
                    ],
                    "MatchAttributes": Object {
                      "DestinationPorts": Array [
                        Object {
                          "FromPort": 22,
                          "ToPort": 22,
                        },
                      ],
                      "Destinations": Array [
                        Object {
                          "AddressDefinition": "10.0.0.0/16",
                        },
                      ],
                      "Protocols": Array [],
                      "SourcePorts": Array [
                        Object {
                          "FromPort": 1024,
                          "ToPort": 65535,
                        },
                      ],
                      "Sources": Array [
                        Object {
                          "AddressDefinition": "10.1.0.0/16",
                        },
                      ],
                    },
                  },
                },
              ],
            },
          },
          "StatefulRuleOptions": Object {
            "RuleOrder": "STRICT_ORDER",
          },
        },
        "RuleGroupName": "TestGroupStateless",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestGroupStateless",
          },
          Object {
            "Key": "someKey",
            "Value": "someValue",
          },
        ],
        "Type": "STATELESS",
      },
      "Type": "AWS::NetworkFirewall::RuleGroup",
    },
  },
}
`;

exports[`Network Firewall Rule Group Stateless Construct(NetworkFirewallPolicy):  Snapshot Test 1`] = `
Object {
  "Resources": Object {
    "TestGroupAF88660E": Object {
      "Properties": Object {
        "Capacity": 100,
        "RuleGroup": Object {
          "RulesSource": Object {
            "StatefulRules": Array [
              Object {
                "Action": "PASS",
                "Header": Object {
                  "Destination": "10.0.0.0/16",
                  "DestinationPort": "ANY",
                  "Direction": "FORWARD",
                  "Protocol": "IP",
                  "Source": "10.1.0.0/16",
                  "SourcePort": "ANY",
                },
                "RuleOptions": Array [
                  Object {
                    "Keyword": "sid",
                    "Settings": Array [
                      "100",
                    ],
                  },
                ],
              },
            ],
          },
          "StatefulRuleOptions": Object {
            "RuleOrder": "STRICT_ORDER",
          },
        },
        "RuleGroupName": "TestGroup",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestGroup",
          },
        ],
        "Type": "STATEFUL",
      },
      "Type": "AWS::NetworkFirewall::RuleGroup",
    },
    "TestGroupStatelessB53F2424": Object {
      "Properties": Object {
        "Capacity": 100,
        "RuleGroup": Object {
          "RuleVariables": Object {
            "IPSets": Object {
              "HOME_NET": Object {
                "Definition": Array [
                  "10.0.0.0/16",
                ],
              },
            },
            "PortSets": Object {
              "HOME_NET": Object {
                "Definition": Array [
                  "80",
                  "443",
                ],
              },
            },
          },
          "RulesSource": Object {
            "StatelessRulesAndCustomActions": Object {
              "CustomActions": Array [
                Object {
                  "ActionDefinition": Object {
                    "PublishMetricAction": Object {
                      "Dimensions": Array [
                        Object {
                          "Value": "CustomValue",
                        },
                      ],
                    },
                  },
                  "ActionName": "CustomAction",
                },
              ],
              "StatelessRules": Array [
                Object {
                  "Priority": 100,
                  "RuleDefinition": Object {
                    "Actions": Array [
                      "aws:pass",
                    ],
                    "MatchAttributes": Object {
                      "DestinationPorts": Array [
                        Object {
                          "FromPort": 22,
                          "ToPort": 22,
                        },
                      ],
                      "Destinations": Array [
                        Object {
                          "AddressDefinition": "10.0.0.0/16",
                        },
                      ],
                      "Protocols": Array [],
                      "SourcePorts": Array [
                        Object {
                          "FromPort": 1024,
                          "ToPort": 65535,
                        },
                      ],
                      "Sources": Array [
                        Object {
                          "AddressDefinition": "10.1.0.0/16",
                        },
                      ],
                    },
                  },
                },
              ],
            },
          },
          "StatefulRuleOptions": Object {
            "RuleOrder": "STRICT_ORDER",
          },
        },
        "RuleGroupName": "TestGroupStateless",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestGroupStateless",
          },
          Object {
            "Key": "someKey",
            "Value": "someValue",
          },
        ],
        "Type": "STATELESS",
      },
      "Type": "AWS::NetworkFirewall::RuleGroup",
    },
  },
}
`;
